notification/views.py

```diff
@@ -29,22 +35,23 @@ def filter_queryset(self, queryset):
 
         return qs
 
-    # Change notification to seen once requested
-    def finalize_response(self, request, *args, **kwargs):
-        result = super().finalize_response(request, *args, **kwargs)
-
-        if not getattr(request, 'child_route', False):
-            queryset = self.paginate_queryset(
-                self.filter_queryset(self.get_queryset()))
-            Notification.objects\
-                .filter(id__in=[q.id for q in queryset])\
-                .update(status=Notification.STATUS_SEEN)
-
-        return result
+    @list_route(permission_classes=[permissions.IsAuthenticated],
```

Comment:
You can also specify serializer_class in this decorator and user `self.get_serializer(data=request.data, many=True)` below. Specifying serializer_class here would make sure that a proper API documentation is auto-generated with correct schema.